TOOLS/odoo-tools.sh

```diff
@@ -1224,23 +1227,124 @@ if [ "$SCRIPT_MODE" = "updatetranslation" ]; then
+    else
+        GREPREGEX="\b${DBNAME}"
+    fi
+    echo "DEBUG: regex: ${GREPREGEX}, dbname: ${DBNAME}"
+    # Todo: Use the correct linux user instead of SU except for full backup. Check rights of psql
+    declare -a DATABASES=($(su - postgres -c "psql --tuples-only -P format=unaligned -c \"SELECT datname FROM pg_database WHERE datname LIKE 'o8_%'\""|grep -o ${GREPREGEX}|sort -u))
+    if [ "x${DATABASES}" = "x" ]; then
+        echo "ERROR: No Database found!"
+    fi
+
+    # ----- HINT: Spezial case STOP Script till odoo Supports this option, service/db.py needs to be extended
+    if [ ${TYPE} = "odoosql" ]; then
+        echo "ERROR: This option is not supported right now "
+        exit 2
+    fi
+    # Todo: check vmware Snapshot how to remote execute the vmware-cmd command with ssh connection to esx server directly, check if the VM is running on this machine
+    # Todo: or find a way of acting through Virtual center server this server has access to the whole cluster and no check on which host a machine is running would be needed
+    # ----- Backup Data for each Instance
+    counter=3
+    for i in "${DATABASES[@]}"; do
+        # Check if we are at PAD CLOUD OR ODOO DB
+
+        # ----- Getting config of database an Parameters
+        INSTANCELOGFILE="${INSTANCE_PATH}/${i}/LOG/IS-BACKUP--${i}--${DATETIME}.log"
+        echo "instancelogfile: ${INSTANCELOGFILE}"
+        INSTANCECONFIGFILE=${INSTANCE_PATH}/${i}/${i}.conf
+        echo "Database Config File --> ${INSTANCECONFIGFILE}" | tee -a ${INSTANCELOGFILE} ${BRANCHLOGFILE}
+        BASEPORT69=($(grep "xmlrpc_port" ${INSTANCECONFIGFILE} | awk '{printf $3;printf "\n"; }'))
+        SUPER_PASSWORD=($(grep "admin_passwd" ${INSTANCECONFIGFILE} | awk '{printf $3;printf "\n"; }'))
+        BACKUPFILE=${INSTANCE_PATH}/${i}/BACKUP/IS-BACKUP--${i}--${DATETIME}
+        echo -e "${DATETIME}: Start ${TYPE} backup for instance ${i}." | tee -a ${INSTANCELOGFILE} ${BRANCHLOGFILE}
+
+        # ----- Now Backing up 3 different ways of Backup Style using collected parameters in the Backup Script above
+        # ----- Backup Style, only for Odoo Databases
+        if [ ${TYPE} = "full" ]; then
+            # Todo: BACKUP all configs relevant from /etc and instance
+            echo "nix"
+        fi
+        # ----- Backup Style, only for Odoozip Databases
+        if [ ${TYPE} = "odoozip" ] || [ ${TYPE} = "full" ]; then # && [ $]; then
+            echo -e $(${INSTANCE_PATH}/TOOLS/db-tools.py -b ${BASEPORT69} -s ${SUPER_PASSWORD} "backup" -d ${i} -f "${BACKUPFILE}_odoo.zip") #-t ${TYPE}
+        echo -e "${DATETIME}: Start ${TYPE} backup for database ${i}." | tee -a ${INSTANCELOGFILE} ${BRANCHLOGFILE}
+            # ----- Check if Backup was at least written to file and File is not zero
+            if ! [ -s "${BACKUPFILE}_odoo.zip" ]; then
+                echo "ERROR: backup was not successfull" | tee -a ${INSTANCELOGFILE} ${BRANCHLOGFILE}
```

Comment:
Woudl be really nice if you woul tell which database was not successful. E.g.: "ERROR: Backup of Database ${i} was not successful!"